File: concentration/cpp/Card.cpp
#include "Card.hpp"
#include <algorithm>

Card::Card(){
    cardNum = -1;
    faceVal = -1;
    suit = "Null";
    name = "Null";
    picture = "Null";
}

Card::Card( int num ){
    cardNum = num;
    faceVal = num % 13 + 1;
    setSuit();
    setName();
    setPicture();
}

void Card::setSuit(){
    if( cardNum < 13 ){
        suit = "Hearts";
    } else if( cardNum < 26 ){
        suit = "Diamonds";
    } else if( cardNum < 39 ){
        suit = "Clubs";
    } else {
        suit = "Spades";
    }
}

void Card::setName(){
    switch ( cardNum % 13) {
        case 12: name = "King"; break;
        case 11: name = "Queen"; break;
        case 10: name = "Jack"; break;
        case 9: name = "Ten"; break;
        case 8: name = "Nine"; break;
        case 7: name = "Eight"; break;
        case 6: name = "Seven"; break;
        case 5: name = "Six"; break;
        case 4: name = "Five"; break;
        case 3: name = "Four"; break;
        case 2: name = "Three"; break;
        case 1: name = "Two"; break;
        case 0: name = "Ace"; break;
        default: name = to_string( cardNum % 13 + 1 ); break;
    }
}

char helpLower( char c ){
    return tolower( c );
}

void Card::setPicture(){
    string n = name;
    string s = suit;

    //transform name to lowercase
    //        input    input stop  output    what doing
    transform( n.begin(), n.end(), n.begin(), &helpLower );

    //transform suit to lower with a lambda function
    // [] capture class balnk means wer are not using any variable from the surrounding scope
    //-> char mean the return type of this function
    transform( s.begin(), s.end(), s.begin(), [](char c) -> char { return tolower( c ); } );

    picture = "cards/" + n + s + ".jpg";
}

string Card::toString() const {
    return name + " of " + suit;
}

File: concentration/Card.php
<?php

class Card{
    private $cardNum;
    private $faceVal;
    private $suit;
    private $name;
    private $picture;

    public static $NORMAL_DECK_SIZE = 52;

    public function __construct( $cardNum ){
        $this->cardNum = $cardNum;
        $this->faceVal = $cardNum % 13 + 1;
        $this->setSuit();
        $this->setName();
        $this->setPicture();
    }

    public function Card(){
        $this->cardNum = -1;
        $this->faceVal = -1;
        $this->suit = "Null";
        $this->name = "Null";
        $this->picture = "Null";
    }

    private function setSuit(){
        if( $this->cardNum < 13 ){
            $this->suit = "Hearts";
        } else if( $this->cardNum < 26 ){
            $this->suit = "Diamonds";
        } else if( $this->cardNum < 39 ){
            $this->suit = "Clubs";
        } else {
            $this->suit = "Spades";
        }
    }

    private function setName(){
        switch ( $this->cardNum % 13) {
            case 12: $this->name = "King"; break;
            case 11: $this->name = "Queen"; break;
            case 10: $this->name = "Jack"; break;
            case 9:  $this->name = "Ten"; break;
            case 8:  $this->name = "Nine"; break;
            case 7:  $this->name = "Eight"; break;
            case 6:  $this->name = "Seven"; break;
            case 5:  $this->name = "Six"; break;
            case 4:  $this->name = "Five"; break;
            case 3:  $this->name = "Four"; break;
            case 2:  $this->name = "Three"; break;
            case 1:  $this->name = "Two"; break;
            case 0:  $this->name = "Ace"; break;
            default: $this->name = to_string( $this->cardNum % 13 + 1 ); break;
        }
    }

    private function setPicture(){
        $n = $this->name;
        $s = $this->suit;
    
        //to lower string
        $n = strtolower( $n );
        $s = strtolower( $s );

        $this->picture = "cards/" . $n . $s . ".jpg";
    }

    public function __toString(){
        return $this->name . " of " . $this->suit;
    }

    public function toHTML() {
        $ss = "<div class=\"card\">" .
            "<img src=\"" . $this->picture . "\" alt=\"" . $this->__toString() . "\" />" .
            "<p>" . $this->__toString() . "</p>".
            "</div>";
        return $ss;
    }


// int getCardNum() const {
// int getFaceVal() const {
// string getSuit() const {
// string getName() const {
// string getPicture() cons
// string toString() const;
// string toHTML() const;
}


?>